### ISSUE TITLE
Is there any point in deleting the "supersampling" item from the ini file?

### ISSUE THREAD

**Issue by BackPonBeauty** [BackPonBeauty/Sega-Model-3-UI-for-20240128-#41]
Hello, 你好，Bonjour，Konichiha，

Previously, I modified the ini file so that unchecking the "supersampling" item would remove it from the file.
[#34](https://github.com/BackPonBeauty/Sega-Model-3-UI-for-20240128-/issues/34)　Issue by [@WarpedPolygon](https://github.com/WarpedPolygon)

When I was looking at the code to update the [Supermodel3-PonMi version](https://github.com/BackPonBeauty/Supermodel3-PonMi), I found that the "supersampling" item was set to 1 by default.

I tested "supersampling".
The image below shows a graph of GPU Usage over time when the "supersampling" item is unchecked, 1, 2, and 3.
Tested with SpikeoutFE on Supermodel3-PonMi (Oct 19, 2024)
The advertisement screen was rotated twice.

![ss](https://github.com/user-attachments/assets/9e273969-b8ba-4826-b3c0-c04af27b1b01)

In my environment, "supersampling=3" caused a drop in frame rate, so I haven't done any further testing.
There is almost no difference in GPU Usage between "uncheck" and "supersampling=1".
I also tried forcibly setting "supersampling=0", but it did not start.
As a result, even if you delete the "supersampling item" from the ini file, it will become "supersampling=1". That's my opinion.

The state is saved, paused, and then the same scene is loaded.

![ss1](https://github.com/user-attachments/assets/8e4ad0cc-56d9-470d-b466-9f2e1881a5dc)
![ss3](https://github.com/user-attachments/assets/21981ae3-7042-44ec-a22b-164fd4ff6a5c)

I would like to hear your opinions on whether or not the function that adds/deletes from the ini file by checking/unchecking the "supersampling item" should be modified.

thx 

Let’s talk about "Back Pon Beauty."

"Back" in English means "the back." "Pon" is the sound of tapping on the back, almost like an onomatopoeia. "Beauty" is often written as 「美」 in Japanese, a character commonly found in girls' names in Japan. When you put it all together, "BackPonBeauty" would translate to "背中ポン美" (Senaka PonMi) in Japanese.

I live alone for various reasons. I often stay up late, writing programs, playing games, drinking, and, well...you know, touching myself.

One night, around 1 a.m., I felt a strange sensation—a light tap on my back. No one else should’ve been there. Startled, I looked around the room, but no one was there. Honestly, if someone had been there, it would’ve been even scarier. I thought it might just be my imagination, but the next night, it happened again. I was tapped on the back, and still, no one was there. By this point, I was starting to suspect some mischievous ghost, like the ones they talk about in English-speaking countries.

The following night, I decided I had to face this "Back Pon" spirit. My plan was simple: if this ghost was going to tap my back, I’d turn around and tap them first. So, I waited. As the clock passed 1 a.m., the atmosphere in the room shifted, and I could sense its presence.

Suddenly, I turned around, closed my eyes in fear, and blindly struck out with a palm heel. "Eek!" came a voice. When I opened my eyes, there stood a girl, about 18 to 20 years old, with pale, translucent skin, clutching her chest. It seemed my strike had landed... well, on her chest.

"What are you touching?!" the girl, whom I started calling "Back Pon," scolded me with a slight pout, glaring at me. Flustered by her cute expression, I blurted out, "You’re adorable."

"N-no, what are you saying?!" she stammered before disappearing.

The next night, she came again, and once more, I sent out a palm heel strike. This time, I hit her... butt. "Eek!" she squeaked again.

"What are you touching this time?!" she huffed, giving me that same pouty look. Again, I thought she was cute but refrained from saying it out loud. However, her face started to twist in anger, so in a panic, I whispered, "Y-you’re cute…"

"Stop saying weird things!" she blushed, before vanishing once more.
Afterward, I called a shaman to cleanse my room and placed salt at the corners. Since then, she hasn’t appeared.

And now, I’m left sleepless once again tonight.

From that day on, I started calling myself "backponbeauty."「背中ポン美」

thanks
PonMi(hentai)

**Comment by leoxxx** [comment]
美（うつく）しい
utsukushii
So 背中ポン美 = Senaka Pon Utsuku
I think this one is so utsukushii!😄
張郃（ちょう こう）：「私の美しさ、まさに真の三國無双！」😀

**Comment by WarpedPolygon** [comment]
In my opinion it's perfect as it is in the UI.

Supersampling is very costly on GPU and I generally can only go to Supersampling=2 before I get performance and or frames dropped with some games (1660 Ti 6GB). x3 does reduce shimmering more than x2 and x3 is what I try to target, but beyond x3 is very much diminishing returns. Not worth the performance hit and is hard to see the difference set past x3.

On my old PC with a 960 1gb I couldn't use SS at all, as you really need 4 or more GBs of VRAM. The more the better as it's brute forcing supersampling.

If supersampling is included in the ini file and set to 0 it doesn't display any image. So having the option to remove it entirely from the ini is absolutely needed for those on lower end hw. Otherwise they cant turn it off.

For me, I think supersampling is great for this emulator and makes games like sega rally 2 look soooooo much better at higher resolutions. No shimmering, yayyy. Keep it the way it is. It's perfect.

**Comment by WarpedPolygon** [comment]
Oh BTW @leoxxx unchecking Supersampling removes it from the ini file for you, and checking adds it to the ini file. No need to manually edit the ini file now.

**Comment by WarpedPolygon** [comment]
@BackPonBeauty  "As a result, even if you delete the "supersampling item" from the ini file, it will become "supersampling=1". That's my opinion."

I don't think that is the case here. On my 960 1gb I couldn't run Supersampling=1 without major performance issues. Setting to 0 gives no image, and removing it from the ini allowed me to play with no performance issues.

Essentially removing it is Supersampling=0, but due to the issue of Supersampling giving no image when set to 0 in the ini file, it needs to be removed to be deactivated.

**Comment by BackPonBeauty** [comment]
@WarpedPolygon
 
Understood.
There doesn't seem to be any particular problems, so I won't make any corrections or changes.

Thank you.

**Comment by BackPonBeauty** [comment]
Continuing from my previous story.

Have you ever felt like someone was watching you?

Sometimes I go to the next town.

I take the bus or train to go for a drink.

When I'm on the train, I feel like someone is watching me.

Women, men, young people, old people, everyone seems to be looking at me.

I don't know if it's my imagination or if our eyes just met by chance, but for some reason they keep looking. I'm very sensitive when our eyes meet, so I usually look away quickly, but they still don't stop looking at me.

Maybe they think I'm a Hollywood actor. (Tehepero)

I suspect they are part of a "mysterious organization". I feel like I'm being watched.

Am I becoming self-conscious?

One day.

I was on the train again.

A faint white translucent figure floated above the head of the woman sitting opposite me.

I couldn't take my eyes off her.

She got off at the next station and gave me a look that asked me, "What are you looking at?"

A white mist followed her.

I think that white mist was a tap on the back.

I was supposed to get off at that station too, but I couldn't move.

There might be something near you, waiting to tap you on the back, too.

PonMi (I'm behind you)

**Comment by BackPonBeauty** [comment]
[@WarpedPolygon](https://github.com/WarpedPolygon)

Deleting the supersampling item from the ini file
and supersampling=1 are equivalent mipmap processing.

When I assert an opinion, I submit supporting documents.
This is evidence.

The GPUUsage was equivalent, as shown in the graph image above.

The difference in still images is also as shown in the image above.

And the code also has an annotation saying "default is 1 which is no aa".

From the official Github below

[Official commit](https://github.com/trzy/Supermodel/commit/c039d08c038fdea8d365920c842e97c09c3a45e7#diff-654aabc47451bf06a0596aa3665e066ce388efac00b6c1db5106b3ddd7a849d7)

```
static unsigned  xOffset, yOffset;      // offset of renderer output within OpenGL viewport
static unsigned  xRes, yRes;            // renderer output resolution (can be smaller than GL viewport)
static unsigned  totalXRes, totalYRes;  // total resolution (the whole GL viewport)
static int aaValue = 1;                 // default is 1 which is no aa
```

```
aaValue = s_runtime_config["Supersampling"].ValueAs<int>();
```

Even if I add '-ss=1' to the command line, that means the mipmap processing is equivalent to not adding the '-ss=n' option.

But that doesn't mean the curse of the "back pat" is lifted.

There's no particular problem, so I won't make any changes.

thanks

PonMi(i....i..will put o...on...Japanese.....curse....ON YOU!!!!!!!!!!!!!!!!!! wahahahahahahahaha.........)

**Comment by BackPonBeauty** [comment]
![image](https://github.com/user-attachments/assets/5a7684e7-e97b-4d96-97a6-21aa5d6c58b2)

**Comment by WarpedPolygon** [comment]
So, I revisited this to see if I was going mad. 

When SS was first introduced Supersampling=1 was actually 1. With the line needing to be removed for SS to not be applied as when set to 0 it doesn't display an image. 

It looks like they changed it so Supersampling=1 is actually SS deactivated. New builds apply Supersampling=1 even if the line isn't in the ini file. Looks like you have confirmed this above, but may need triple checking.

My question is, if Supersampling=1 is actually OFF, should the UI reflect that? 
So Supersampling=1 shows as OFF in the UI, with Supersampling=2 actually being Supersampling=1, Supersampling=3 actually being Supersampling=2 etc etc.

Am I explaining this right? lol

**Comment by BackPonBeauty** [comment]
@WarpedPolygon 

In Supermodel3, Supersampling typically refers to SSAA (Super-Sampling Anti-Aliasing), a technique that renders an image at a higher resolution and then downsamples it to smooth out jagged edges. It is usually set in multiples of 2.

When Supersampling was first introduced in Supermodel3, the default value was 1, meaning that the rendering resolution remains the same as the original. If set to 0, it results in 0 times the resolution, causing nothing to be rendered and leading to a crash.

Additionally, if you remove the Supersampling entry from the ini file, the application will not find the value and will instead apply the default setting, which is 1. Therefore, setting Supersampling=1 or removing the entry from the ini file yields the same result.

If the UI interprets Supersampling=1 as "off," then Supersampling=2 might correspond to an actual setting of 2x, Supersampling=3 to 6x, and Supersampling=4 to 8x. Since anti-aliasing is typically configured in multiples of 2, using odd values may result in non-standard scaling factors. This behavior depends on internal processing and potential changes in the specifications, so it should be carefully verified.

In conclusion, the valid range for Supermodel3's Supersampling setting is from 1 to 8, with 1 being the default.
You may find this difficult to understand, but this is the specification of Supermodel3’s ini file, as well as my UI design. I have no intention of changing it.

**Comment by BackPonBeauty** [comment]
@WarpedPolygon 

I've thought about it carefully.
I think it might be a good idea to change it.
Please give me some time.
I'll include it in the next update.

Thank you.

**Comment by WarpedPolygon** [comment]
Great. Thanks for hearing me out. It's not that I don't understand, just that I feel that most people will think 
Supersampling=1, is actually on and set to 1. Not off. And the other corresponding numbers should be correctly displayed irrespective of how they 'appear' in the ini.

Really it's figuring out what the supermodel team have done regarding the Supersampling numbers. 

So, if 1 is OFF, I would assume 2 is 1, 3 is 2, 4 is 3, and so on. Mathematically, if 1 = 0 then the scale should go to 9 in the ini to accommodate for that. But the scale goes to 8 at this current time.

I think they may have just set 1=OFF,.......... and 2 is actually 2, technically skipping 1 entirely. Just my theory

**Comment by BackPonBeauty** [comment]
@WarpedPolygon 

Your statement, **"What is Supersampling=2 actually set to if 1 is off?"** is not a valid premise for this discussion.  

First, you should take the time to understand **Supersampling Anti-Aliasing (SSAA)**:  
🔗 [https://en.wikipedia.org/wiki/Supersampling](https://en.wikipedia.org/wiki/Supersampling)  

In the `ini` file, **Supersampling=1 does not mean SSAA=OFF, nor does it mean SSAA=0. It is also not skipped. It represents SSAAx1.**  

SSAA stands for **Supersampling Anti-Aliasing**, which is one of several anti-aliasing techniques. Other methods include **MSAA, FAA, CFAA, and FXAA**.  

Thus, **Supersampling=2 corresponds to SSAAx2**.  
There is no contradiction between the values set in the `ini` file and the SSAAx _n_ values.  

If I were to make a modification to the UI, I would change the current Supersampling slider display from 1–8 to x1–x8.

I would simply display 1 as x1.

It is neither OFF nor 0—1 will be displayed as x1.

So, in the end, how do you think this should be correctly expressed?

**Comment by WarpedPolygon** [comment]
Perhaps, I have fundamentally misunderstood.

It is fine as it is. the x1, x2 etc isn't necessary. Apologies for wasting your time.

**Comment by BackPonBeauty** [comment]
Hey hey hey hey

This guy deleted his comment and ran away.

I'm reposting the post he deleted here.

```
So, I revisited this to see if I was going mad.

When SS was first introduced Supersampling=1 was actually 1. With the line needing to be removed for SS to not be applied as when set to 0 it doesn't display an image.

It looks like they changed it so Supersampling=1 is actually SS deactivated. New builds apply Supersampling=1 even if the line isn't in the ini file. Looks like you have confirmed this above, but may need triple checking.

My question is, if Supersampling=1 is actually OFF, should the UI reflect that?
So Supersampling=1 shows as OFF in the UI, with Supersampling=2 actually being Supersampling=1, Supersampling=3 actually being Supersampling=2 etc etc.

Am I explaining this right? lol
```

The next post he posted

```
Great. Thanks for hearing me out. It's not that I don't understand, just that I feel that most people will think
Supersampling=1, is actually on and set to 1. Not off. And the other corresponding numbers should be correctly displayed irrespective of how they 'appear' in the ini.

Really it's figuring out what the supermodel team have done regarding the Supersampling numbers.

So, if 1 is OFF, I would assume 2 is 1, 3 is 2, 4 is 3, and so on. Mathematically, if 1 = 0 then the scale should go to 9 in the ini to accommodate for that. But the scale goes to 8 at this current time.

I think they may have just set 1=OFF,.......... and 2 is actually 2, technically skipping 1 entirely. Just my theory
```


You'd better learn internet literacy and etiquette.

If you delete a comment that's inconvenient for you, the same stupid question will be repeated over and over by the same fool.

Such comments reassure even more stupid people that there are people out there who are stupider than you.

So don't delete your comments.

If you do it again,
some kind of penalty will have to be given.

Do you want to be blocked?

Also, when commenting, if you claim some kind of bug or problem, please provide evidence. At least have some knowledge about emulators and understand the terms and their meanings before you comment.

And provide a way to reproduce the bug or problem or verification results.

I showed above that Supersampling = 1 is the default setting. There is also an image comparing GPU usage, and I also showed the relevant code.
Please provide evidence for your claim that "some kind of problem will occur if the supersampling value is not removed from the ini file."
Why don't you try filming a verification video and uploading it to YouTube? It's easy.

You should show the circumstances and reasons why you assumed "Supersampling = OFF". If it's just an assumption, that's fine.

People sometimes make mistakes because of assumptions.

Everyone does.

But if you notice it and listen to what others say, and if your claim is wrong, accept it honestly. You just need to understand.

You don't need to apologize to me.

I just shared some knowledge with you.

I don't know if you understood.

But even though I patiently answered your stupid questions and comments that even an 8-year-old could answer many times, I cannot accept that you deleted the comment you posted.

If you're a respectable adult, you should show evidence if you make a claim.
It's common sense for a working adult. If you're an 8-year-old, it can't be helped.

It's a waste of time, so I'm going to close this issue using my authority.

I won't make any corrections.
